aws/fake/cf.go

```diff
@@ -0,0 +1,97 @@
+package fake
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/cloudformation/cloudformationiface"
+)
+
+type CfMockOutputs struct {
+	DescribeStackPages          *ApiResponse
+	DescribeStacks              *ApiResponse
+	CreateStack                 *ApiResponse
+	UpdateStack                 *ApiResponse
+	DeleteStack                 *ApiResponse
+	UpdateTerminationProtection *ApiResponse
+}
+
+type MockCloudFormationClient struct {
+	cloudformationiface.CloudFormationAPI
+	lastGeneratedTemplate string
+	Outputs               CfMockOutputs
+}
+
+func (m *MockCloudFormationClient) GetLastGeneratedTemplate() string {
+	return m.lastGeneratedTemplate
+}
+
+func (m *MockCloudFormationClient) DescribeStacksPages(in *cloudformation.DescribeStacksInput, fn func(*cloudformation.DescribeStacksOutput, bool) bool) (err error) {
+	if m.Outputs.DescribeStackPages != nil {
+		err = m.Outputs.DescribeStackPages.err
+	}
+	if err != nil {
+		return
+	}
+
+	if out, ok := m.Outputs.DescribeStacks.response.(*cloudformation.DescribeStacksOutput); ok {
+		fn(out, true)
+	}
+	if m.Outputs.DescribeStacks != nil {
+		err = m.Outputs.DescribeStacks.err
+	}
+
+	return
+}
+
+func (m *MockCloudFormationClient) DescribeStacks(in *cloudformation.DescribeStacksInput) (*cloudformation.DescribeStacksOutput, error) {
+	if out, ok := m.Outputs.DescribeStacks.response.(*cloudformation.DescribeStacksOutput); ok {
+		return out, m.Outputs.DescribeStacks.err
+	}
+	return nil, m.Outputs.DescribeStacks.err
+}
+
+func (m *MockCloudFormationClient) CreateStack(params *cloudformation.CreateStackInput) (*cloudformation.CreateStackOutput, error) {
+	m.lastGeneratedTemplate = *params.TemplateBody
+	if out, ok := m.Outputs.CreateStack.response.(*cloudformation.CreateStackOutput); ok {
+		return out, m.Outputs.CreateStack.err
```

Comment:
this is the happy path and it should be on the left for readability.
So as early wrote, please refactor all functions to have the Go style. Thanks